src/praisonai-agents/praisonaiagents/agent/agent.py

```
@@ -526,6 +526,44 @@ def generate_task(self) -> 'Task':
+            return arguments
+        
+        try:
+            sig = inspect.signature(func)
+            casted_args = {}
+            
+            for param_name, arg_value in arguments.items():
+                if param_name in sig.parameters:
+                    param = sig.parameters[param_name]
+                    if param.annotation != inspect.Parameter.empty:
+                        # Handle common type conversions
+                        if param.annotation == int and isinstance(arg_value, (str, float)):
```

Ruff (0.11.9)

Use `is` and `is not` for type comparisons, or `isinstance()` for isinstance checks

(E721)